libgalois/include/katana/Timer.h

```diff
@@ -104,17 +104,24 @@ class CondStatTimer<false> {
   uint64_t get_usec() const { return 0; }
 };
 
-template <typename F>
-void
-timeThis(const F& f, const char* const name) {
-  StatTimer t("Time", name);
+//! Start and stop a TimeAccumulator in the scope of the guard.
+template <typename T>
+class [[nodiscard]] TimerGuard {
+  // The template parameter is required because TimeAccumulator::start and stop
```

Comment:
Maybe it is just me, but I don't think it is too unexpected that TimerGuard is a templated class, unless you mean that TimerGuard really should only be used with TimeAccumulator?

Even in that case, I think `TimerGuard` being ambivalent to being used with `TimeAccumulator` versus any class that has `start` and `stop` seems reasonable.

Reply:
That's fair. I just wanted to document the design choice since the class very much is designed to be used with `TimeAccumulator`. Though as you say it's not actually specific to that.